ioc_finder/ioc_finder.py

```diff
@@ -418,105 +451,148 @@ def find_iocs(  # noqa: CCR001 pylint: disable=R0912,R0915
     original_text = text
 
     # urls
-    iocs['urls'] = parse_urls(text, parse_urls_without_scheme=parse_urls_without_scheme)
-    if not parse_domain_from_url and not parse_from_url_path:
-        text = _remove_items(iocs['urls'], text)
-    elif not parse_domain_from_url:
-        text = _percent_decode_url(iocs['urls'], text)
-        text = _remove_url_domain_name(iocs['urls'], text)
-    elif not parse_from_url_path:
-        text = _percent_decode_url(iocs['urls'], text)
-        text = _remove_url_paths(iocs['urls'], text)
-    else:
-        text = _percent_decode_url(iocs['urls'], text)
+    if "urls" in included_ioc_types:
+        iocs['urls'] = parse_urls(text, parse_urls_without_scheme=parse_urls_without_scheme)
+        if not parse_domain_from_url and not parse_from_url_path:
+            text = _remove_items(iocs['urls'], text)
+        elif not parse_domain_from_url:
```

Comment:
Theoretically, this conditional statement could be more descriptive to only run if 'domains' and [other ioc types which can be a url domain name](https://github.com/fhightower/ioc-finder/blob/e531ca13b77df941202afa8e96c61082bdcf6cad/ioc_finder/ioc_grammars.py#L112) are in included_data_types, but I'm not going to implement this now as the complexity of implementation is not worth the execution time that would be saved.